AVX512BW detection is buggy

This works as expected:
``` 
./configure # OR ./configure CPPFLAGS='-mavx512bw'
[...]
Target CPU ................................. x86_64 AVX512BW, 64-bit LE
AES-NI support ............................. depends on OpenSSL
Target OS .................................. linux-gnu
Cross compiling ............................ no
Legacy arch header ......................... x86-64.h
```

But this fails:
```
./configure --disable-native-tests CPPFLAGS='-mavx512bw'
[...]
checking for MMX... yes
checking for SSE2... yes
checking for SSSE3... yes
checking for SSE4.1... yes
checking for SSE4.2... yes
checking for AVX... yes
checking for XOP... no
checking for AVX2... yes
checking for AVX512F... yes
checking for AVX512BW... no
[...]
Target CPU ................................. x86_64 AVX512F, 64-bit LE
AES-NI support ............................. depends on OpenSSL
Target OS .................................. linux-gnu
Cross compiling ............................ no
Legacy arch header ......................... x86-64.h
```

```
configure:11595: checking for AVX512BW
configure:11607: gcc -o conftest -m64 -g -O2 -I/usr/local/include   -O0 -mavx512bw -I/usr/local/include -m64  -L/usr/local/lib -L/usr/local/lib64 -L/usr/lib64 -L/lib64 conftest.c  >&5
conftest.c: In function 'main':
conftest.c:30:9: error: can't convert a vector of type '__m512i' {aka '__vector(8) long long int'} to type 'int' which has different size
         int main(){__m512i t, t1;*((long long*)&t)=1;t1=t;t=_mm512_slli_epi16(t1,t);if((*(long long*)&t)==88)printf(".");exit(0);}
         ^~~
configure:11607: $? = 1
configure: failed program was:
```

--- a/src/configure
+++ b/src/configure
@@ -11600,7 +11600,7 @@ $as_echo_n "checking for AVX512BW... " >&6; }
     #include <immintrin.h>
         #include <stdio.h>
         extern void exit(int);
-        int main(){__m512i t, t1;*((long long*)&t)=1;t1=t;t=_mm512_slli_epi16(t1,t);if((*(long long*)&t)==88)printf(".");exit(0);}
+        int main(){__m512i t, t1;*((long long*)&t)=1;t1=t;t=_mm512_slli_epi32(t1,t);if((*(long long*)&t)==88)printf(".");exit(0);}
 
 
 _ACEOF

--- a/src/m4/jtr_x86_logic.m4
+++ b/src/m4/jtr_x86_logic.m4
@@ -469,7 +469,7 @@ dnl ======================================================================
       [[#include <immintrin.h>
         #include <stdio.h>
         extern void exit(int);
-        int main(){__m512i t, t1;*((long long*)&t)=1;t1=t;t=_mm512_slli_epi16(t1,t);if((*(long long*)&t)==88)printf(".");exit(0);}]]
+        int main(){__m512i t, t1;*((long long*)&t)=1;t1=t;t=_mm512_slli_epi32(t1,t);if((*(long long*)&t)==88)printf(".");exit(0);}]]
     )]
     ,[CPU_BEST_FLAGS="-mavx512bw"]
      [SIMD_NAME="AVX512BW"]

